docs/docs/module_guides/indexing/lpg_index_guide.md

```diff
@@ -0,0 +1,573 @@
+from qdrant_client import QdrantClient, AsyncQdrantClient
+
+vector_store = QdrantVectorStore(
+    "graph_collection",
+    client=QdrantClient(...),
+    aclient=AsyncQdrantClient(...),
+)
+
+graph_store = Neo4jLPGStore(...)
+
+# creates an index
+index = PropertyGraphIndex.from_documents(
+    documents,
+    property_graph_store=graph_store,
+    vector_store=vector_store,
+    embed_kg_nodes=True,
+)
+
+# load from existing graph/vector store
+index = PropertyGraphIndex.from_existing(
+    property_graph_store=graph_store,
+    vector_store=vector_store,
+    embed_kg_nodes=True,
+)
+```
+
+### Using the LPG Graph Store Directly
+
+The base storage class for property graphs is the `PropertyGraphStore`. These property graph stores are constructured using different types of `LabeledNode` objects, and connected using `Relation` objects.
+
+We can create these ourselves, and also insert ourselves!
+
+```python
+from llama_index.core.graph_stores import (
+    SimplePropertyGraphStore,
+    EntityNode,
+    Relation,
+)
+from llama_index.core.schema import TextNode
+
+graph_store = SimplePropertyGraphStore()
+
+entities = [
+    EntityNode(name="llama", label="ANIMAL", properties={"key": "val"}),
+    EntityNode(name="index", label="THING", properties={"key": "val"}),
+]
+
+relations = [
+    Relation(
+        label="HAS",
+        source_id=entities[0].id,
+        target_id=entities[1].id,
+        properties={},
+    )
+]
+
+graph_store.upsert_nodes(entities)
+graph_store.upsert_relations(relations)
+
+# optionally, we can also insert text chunks
+source_chunk = TextNode(id_="source", text="My llama has an index.")
+
+# create relation for each of our entities
+source_relations = [
+    Relation(
+        label="HAS_SOURCE",
+        source_id=entities[0].id,
+        target_id="source",
+    ),
+    Relation(
+        label="HAS_SOURCE",
+        source_id=entities[1].id,
+        target_id="source",
+    ),
+]
+graph_store.upsert_llama_nodes([source_chunk])
+graph_store.upsert_relations(source_relations)
+```
+
+Other helpful methods on the graph store include:
+- `graph_store.get(ids=[])` - gets nodes based on ids
+- `graph_store.get(properties={"key": "val"})` - gets nodes based on matching properties
+- `graph_store.get_rel_map([entity_node], depth=2)` - gets triples up to a certain depth
+- `graph_store.get_llama_nodes(['id1'])` - gets the original text nodes
+- `graph_store.delete(ids=['id1'])` - delete based on ids
+- `graph_store.delete(properties={"key": "val"})` - delete based on properties
+- `graph_store.structured_query("<cypher query>")` - runs a cypher query (assuming the graph store supports it)
+
+In addition `a` versions exist for all of these for async support (i.e. `aget`, `adelete`, etc.).
+
+## Advanced Customization
+
+As with all components in LlamaIndex, you can sub-class modules and customize things to work exactly as you need, or try out new ideas and research new modules!
+
+### Sub-Classing Extractors
```

Comment:
this and custom retriever probably worth their own notebooks (i can help with this)